### PR TITLE
chore(raiko-lib): auto logging

### PR DESCRIPTION
**Pull request by 35359595**
* automatic logging if `RUST_LOG` env not present is set to `info` level;
* `println` logs replaced with `tracing::debug`;
* logging levels demoted to `debug` on most chatty outputs;
* install script sources sp1 not only on CI but also if not found in current $PATH;

### CODE REVIEW

**Comment by dantaik** [comment]
you have to fix your PR title first, something like "chore(project_name): verb ..."

File: host/src/main.rs [hunk 1]
```diff
@@ -11,6 +10,10 @@ use tracing_subscriber::FmtSubscriber;
 
 #[tokio::main]
 async fn main() -> HostResult<()> {
+    // defailt `info` level if nothing else defined
+    if env::var("RUST_LOG").is_err() {
+        env::set_var("RUST_LOG", "info")
+    }
```
Comment (Brechtpd) [inline host/src/main.rs:16]: Seems already set to the default value here: https://github.com/taikoxyz/raiko/blob/a1ee3ae56704658807ced7aa5fef10b56df3986a/host/src/lib.rs#L97-L100 @petarvujovic98?

File: lib/src/lib.rs [hunk 2]
```diff
@@ -124,15 +126,15 @@ impl Measurement {
...
+    debug!(
         "{title}{}.{:03} seconds",
         duration.as_secs(),
         duration.subsec_millis()
     );
 }
 
 pub fn inplace_print(title: &str) {
-    print!("\r{title}");
+    debug!("\r{title}");
```
Comment (Brechtpd) [inline lib/src/lib.rs:137]: breaks the inplace printing for the nice single line updating of progress though... :(

**Review by Brechtpd — COMMENTED** [review]
Some good changes! 

But feels like `info` now prints very little, and `debug` prints so much stuff it's hard to find anything back, so I think we need to tweak things some more so that the default log level still prints out a sufficient amount of useful information. Also seems like https://github.com/taikoxyz/raiko/blob/a1ee3ae56704658807ced7aa5fef10b56df3986a/lib/src/builder/execute.rs#L286 needs to changed to `trace` level now if we want to use the `debug` level more because that one is very spammy and normally not useful.

I think it's okay to print some stuff using `println` for stuff that is mostly just important while you're waiting on the block, like the transaction progress or the data fetching progress. That stuff is not so important for the logs, but very useful while you're staring at it and wondering how long something will still take to finish.

Currently the block execution though is very messy because of the optimistic execution still printing a lot of stuff, I think to solve that we could  not print/log anything when `is_optimistic` is true in `execute_transactions`. Or something like that. We could also have a custom logger trait that people could set but that seems overkill.

File: host/src/main.rs (hunk 1 above)
Comment (petarvujovic98) [reply host/src/main.rs:16]: Yes, it is already the default and used here https://github.com/taikoxyz/raiko/blob/a1ee3ae56704658807ced7aa5fef10b56df3986a/host/src/main.rs#L32-L38

**Comment by petarvujovic98** [comment]
Good stuff here @35359595. Some of the `println` usage for progress/waiting feedback could get hidden behind a `#[cfg(debug_assertions]` so it's only visible when running in dev mode (e.g. people looking at the screen for progress), not suitable for tracing as that data doesn't help with either metrics or debugging after the fact.a

File: host/src/main.rs (hunk 1 above)
Comment (35359595) [reply host/src/main.rs:16]: I dont believe that approach works (or is triggered). Will double check.

Comment (35359595) [reply host/src/main.rs]: it does work, some logs were trying to output before init was happening. re-shuffled and reverted my init. Thanks.

**Comment by johntaiko** [comment]
How is the progress going here? 
image
It has some negative impacts on our production environment, and we hope it can be resolved ASAP :)

File: lib/src/lib.rs [hunk 3]
```diff
@@ -134,13 +134,19 @@ pub fn print_duration(title: &str, duration: time::Duration) {
 }
 
 pub fn inplace_print(title: &str) {
-    print!("\r{title}");
-    #[cfg(all(feature = "std", debug_assertions))]
-    io::stdout().flush().unwrap();
+    if cfg!(debug_assertions) {
+        print!("\r\n{title}");
```
Comment (johntaiko) [inline lib/src/lib.rs:138]: `\r\n` new line for both OS(macOS, unix, windows)

File: core/src/lib.rs [hunk 4]
```diff
@@ -68,6 +67,7 @@ impl Raiko {
             Ok((header, _mpt_node)) => {
                 info!("Verifying final state using provider data ...");
                 info!("Final block hash derived successfully. {}", header.hash());
+                debug!("Final block header derived successfully. {header:?}");
```
Comment (petarvujovic98) [inline core/src/lib.rs:70]: Why both `debug` and `info` here?

File: lib/src/builder/execute.rs
Comment (petarvujovic98) [inline lib/src/builder/execute.rs:1]: Why are the measurements removed here?

**Review by petarvujovic98 — CHANGES_REQUESTED** [review]

File: lib/Cargo.toml
[the remark this answers is not in the document]
Comment (35359595) [reply lib/Cargo.toml:1]: Which exact piece is of concern? I believe `println` is being part of `std` too.

File: lib/src/builder/execute.rs
[the remark this answers is not in the document]
Comment (35359595) [reply lib/src/builder/execute.rs:1]: not sure which exact code is this comment related to? It's instant `Outdated`...

File: core/src/lib.rs (hunk 4 above)
Comment (35359595) [reply core/src/lib.rs:70]: info for just `hash` and debug for entire `header`

File: lib/src/builder/execute.rs
Comment (35359595) [reply lib/src/builder/execute.rs:1]: I'm not sure. @Brechtpd removed those in this commit: https://github.com/taikoxyz/raiko/pull/219/commits/549ddc3e1cde9988cd34f4d66df668c417754d6e

File: core/src/lib.rs (hunk 4 above)
Comment (petarvujovic98) [reply core/src/lib.rs:70]: I see that, but the entire header is both `info` and `debug` printed as you can see here:
https://github.com/taikoxyz/raiko/blob/1e9987c1beb0fc70042ccfdae3cdc2bc3a6178e5/core/src/lib.rs#L70-L71

Comment (35359595) [reply core/src/lib.rs:70]: oh, missed 71... probably just replaced `println!`. will clean up.

**Review by petarvujovic98 — APPROVED** [review]

**Pull request merged**